### PR TITLE
Hide sidebar goals with no data for countries.

### PR DESCRIPTION
**Pull request by keyurva**
[dynamic-sidebar.webm](https://github.com/datacommonsorg/website/assets/1221814/fb43830f-91d2-447d-ab21-f81140b2e86a)

### CODE REVIEW

File: experimental/sdg-tracker/src/components/goals/Goals.tsx [hunk 1]
```diff
@@ -65,7 +66,7 @@ const Goals: React.FC = () => {
               hidePlaceSearch={true}
               variableDcids={[variableDcid]}
               placeDcids={[EARTH_PLACE_DCID]}
-              setPlaceDcid={() => {}}
+              setPlaceDcid={() => { }}
```
Comment (dwnoble) [inline experimental/sdg-tracker/src/components/goals/Goals.tsx:69]: Can you install the vscode "prettier" plugin for code formatting (and enable "run on save") - that will keep code formatting like here consistent across everyone
https://www.robinwieruch.de/how-to-use-prettier-vscode/

File: experimental/sdg-tracker/src/utils/DataCommonsClient.ts [hunk 2]
```diff
@@ -86,5 +86,17 @@ class DataCommonsClient {
...
+  async existence(payload: BulkObservationExistenceRequest): Promise<any> {
+    const url = `${this.apiRoot}/api/observation/existence`;
+    const response = await fetch(url, {
+      method: "POST",
+      headers: {
+        "Content-Type": "application/json",
+      },
+      body: JSON.stringify(payload),
+    });
+    return (await response.json()) as any;
```
[the remark this answers is not in the document]
Comment (keyurva) [reply experimental/sdg-tracker/src/utils/DataCommonsClient.ts]: Done.

File: experimental/sdg-tracker/src/components/goals/Goals.tsx (hunk 1 above)
Comment (keyurva) [reply experimental/sdg-tracker/src/components/goals/Goals.tsx]: Done.

File: experimental/sdg-tracker/src/state/index.ts [hunk 3]
```diff
@@ -288,6 +302,51 @@ const appActions: AppActions = {
...
+      }
+      if (!allTopicDcids || allTopicDcids.length === 0) {
+        return [];
+      }
+
+      const request: BulkObservationExistenceRequest = {
+        entities: [placeDcid],
+        variables: allTopicDcids,
+      };
+      const response = await dataCommonsClient.existence(request);
```
[the remark this answers is not in the document]
Comment (keyurva) [reply experimental/sdg-tracker/src/state/index.ts]: Done.

**Review by dwnoble — APPROVED** [review]
Nice!

**Pull request merged**